### DISCUSSION TITLE
bulk_extractor V2.0.0 RELEASE

### DISCUSSION THREAD

**Discussion by simsong** [simsong/bulk_extractor#342] [Announcements]
Release 2.0.0 of `bulk_extractor`, a high-performance digital forensics tool that works like a "find evidence" button, pulling actionable intelligence out of disk images, files, memory dumps, network traffic, and just about anything else.

<hr /><em>This discussion was created from the release <a href='https://github.com/simsong/bulk_extractor/releases/tag/v2.0.0'>bulk_extractor V2.0.0 RELEASE</a>.</em>

**Comment by kieranjol** [comment]
Stupid question, is this purely the CLI version, and is BE Viewer and the bulk extractor GUI a separate project?

**Reply by simsong** [reply]
Thanks for the question. BE2.0.0 is a linux/macos validated source code release. Versions of the beta were tested under win64 with MinGW, but there is no installer. It is a drop-in replacement for BE1.5, so you can drop it in to an existing release of BEViewer and it should work. I have not updated the Windows installer either, though.

Based on your comment, I have created these issues:
https://github.com/simsong/bulk_extractor/issues/343
https://github.com/simsong/bulk_extractor/issues/344
https://github.com/simsong/bulk_extractor/issues/345

Do you want to help?

**Reply by kieranjol** [reply]
Oh absolutely! I’ve been using bulk extractor in the National Library of Ireland over the last year, so if I interpret you correctly, will I test the windows CLI version of 2.0 and view the reports in BE viewer on windows? I’m not sure how to get beviewer to use v2.0 to create reports, but I’ll do my best to help out with any testing that you might need. Apologies if I’ve misunderstood you.

**Reply by simsong** [reply]
It's great to have you as a user!
If you are running BE on Windows, it needs to be cross-compiled from Fedora. I am happy to do that for you, but I won't be able to do it for the next week. However, it would be great to have you do the testing in a complex environment. Are you using BE with BitCurator? Please send me your contact info so that we can stay in touch.

**Reply by kieranjol** [reply]
We mostly use it within BitCurator via VirtualBox, but it's also used on Windows as well. I also use it within https://github.com/tw4l/brunnhilde on both Windows and BitCurator. I'll shot you an email now!

**Reply by simsong** [reply]
Thanks. I'm not in touch with the current BitCurator developers. Please ask them to talk to me also so that we can coordinate their upgrade. 

In particular, 2.0.0 adds no new decompression or decoding algorithms. There is a pressing need to add them, and it would be reasonable to discuss an approach for funding the development effort. The BE2.0 update I did entirely as volunteer work.

**Reply by kamwoods** [reply]
Apologies for resurrecting an old(er) discussion. BitCurator has been significantly revised thanks to volunteer contributions; the latest releases (now at https://github.com/bitcurator/bitcurator-salt) install BE2.0 by default in either of the most recent LTS releases of Ubuntu, 20.04 and 22.04. We've also conducted some testing to validate that BE2.0 in BitCurator on both LTS platforms plays nicely with included archives-specific tools such as the most current releases of Brunnhilde and Bulk Reviewer.

**Reply by simsong** [reply]
This is great news! Thanks for the update. I look forward to working more closely with the BitCurator team in the future!

**Comment by ghost** [comment]
Hi there, 

I download the tar.gz file, unpack it, run the ./configure command and get the first error: "configure: error: CommonCrypto, SSL/OpenSSL, or gcrypto support required".

If I still proceed with the make command, it won't start because the relevant file is missing.

Anything I could do? 

I am running Ubuntu 20.04 LTS in Virtualbox.

Thank you!

**Comment by simsong** [comment]
Yes, you can install  SSL/OpenSSL or gcrypto
